floodlight/io/datasets.py

```diff
@@ -265,3 +260,373 @@ def get_pitch(self) -> Pitch:
+                f"{self._STATSBOMB_SCHEMA}://"
+                f"{self._STATSBOMB_BASE_URL}/"
+                f"{self._STATSBOMB_THREESIXTY_FOLDERNAME}/"
+                f"{str(mID)}"
+                f"{self._STATSBOMB_FILE_EXT}"
+            )
+            try:
+                data = download_from_url(threesixty_host_url)
+                with open(filepath_threesixty, "wb") as binary_file:
+                    binary_file.write(data)
+            except HTTPError:
+                filepath_threesixty = None
+
+        # read events from file
+        (home_ht1, home_ht2, away_ht1, away_ht2,) = read_open_statsbomb_event_data_json(
+            filepath_events, filepath_matches, filepath_threesixty
+        )
+        event_objects = (home_ht1, home_ht2, away_ht1, away_ht2)
+
+        return event_objects
+
+    @staticmethod
+    def get_pitch() -> Pitch:
+        """Returns a Pitch object corresponding to the StatsBomb Dataset."""
+        return Pitch.from_template("statsbomb", sport="football")
+
+    def _read_competition_season_and_match_links_from_files(self):
+        """Creates the dictionaries containing data links between competition, season,
+        and matches to the respective cIDs, sIDs, and mIDs for all available matches and
+        every competition and season in the StatsBomb dataset.
+        """
+        # updates on competition level
+        competition_info = pd.read_json(self.filepath_competitions)
+        cIDs = competition_info["competition_id"].unique()
+        competitions = competition_info["competition_name"].unique()
+        self.links_competition_to_cID.update(
+            {competition: cIDs[i] for i, competition in enumerate(competitions)}
+        )
+        self.links_season_to_sID.update(
+            {competition: {} for competition in competitions}
+        )
+        self.links_match_to_mID.update(
+            {competition: {} for competition in competitions}
+        )
+
+        # loop
+        for _, single_season in competition_info.iterrows():
+            # update on season level
+            cID = single_season["competition_id"]
+            competition = single_season["competition_name"]
+            sID = single_season["season_id"]
+            season = single_season["season_name"]
+            self.links_season_to_sID[competition].update({season: sID})
+
+            # update on match level
+            with open(
+                os.path.join(
+                    os.path.join(self._matches_data_dir, str(cID)),
+                    str(sID) + self._STATSBOMB_FILE_EXT,
+                ),
+                "rb",
+            ) as matches_file:
+                matches_info = json.load(matches_file)
+            self.links_match_to_mID[competition][season] = {
+                f"{info['home_team']['home_team_name']} "
+                f"vs. "
+                f"{info['away_team']['away_team_name']}": info["match_id"]
+                for info in matches_info
+            }
```

Comment:
please try to be more explicit here or provide more inline comments, its really hard to understand what this function actually does